x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -7,64 +7,152 @@
+import { TimeRangeBounds } from '../../util/time_buckets';
+import { MlKibanaReactContextValue } from '../../contexts/kibana';
+// @ts-ignore
+import { getFieldTypeFromMapping } from '../../services/mapping_service';
+import { AnomaliesTableRecord } from '../../../../common/types/anomalies';
+
+interface LinksMenuProps {
+  anomaly: AnomaliesTableRecord;
+  bounds: TimeRangeBounds;
+  showViewSeriesLink: boolean;
+  isAggregatedData: boolean;
+  interval: 'day' | 'hour' | 'second';
+  showRuleEditorFlyout: (anomaly: AnomaliesTableRecord) => void;
+  kibana: MlKibanaReactContextValue;
+}
+
+export const LinksMenuUI = (props: LinksMenuProps) => {
+  const [isPopoverOpen, setPopoverOpen] = useState(false);
+  const [openInDiscoverUrl, setOpenInDiscoverUrl] = useState<string | undefined>();
+
+  useEffect(() => {
+    let unmounted = false;
+    const generateDiscoverUrl = async () => {
+      const {
+        services: { share },
+      } = props.kibana;
+      const discoverLocator = share.url.locators.get('DISCOVER_APP_LOCATOR');
+
+      if (!discoverLocator) {
+        // eslint-disable-next-line no-console
+        console.error('No locator for Discover detected');
+        return;
+      }
+
+      const job = mlJobService.getJob(props.anomaly.jobId);
+
+      const index = job.datafeed_config.indices[0];
+      const interval = props.interval;
+      const dataViewId = (await getDataViewIdFromName(index)) || index;
+      const record = props.anomaly.source;
+
+      const earliestMoment = moment(record.timestamp).startOf(interval);
+      if (interval === 'hour') {
+        // Start from the previous hour.
+        earliestMoment.subtract(1, 'h');
+      }
+      let latestMoment = moment(record.timestamp).add(record.bucket_span, 's');
+      if (props.isAggregatedData === true) {
+        latestMoment = moment(record.timestamp).endOf(interval);
+        if (interval === 'hour') {
+          // Show to the end of the next hour.
+          latestMoment.add(1, 'h'); // e.g. 2016-02-08T18:59:59.999Z
+        }
+      }
+      const from = timeFormatter(earliestMoment.unix() * 1000); // e.g. 2016-02-08T16:00:00.000Z
+      const to = timeFormatter(latestMoment.unix() * 1000);
+
+      let kqlQuery = '';
+
+      if (record.influencers) {
+        kqlQuery = record.influencers
+          .map(
+            (influencer) =>
+              `${escapeForElasticsearchQuery(influencer.influencer_field_name)}:"${
+                influencer.influencer_field_values[0] ?? ''
+              }"`
+          )
+          .join(' AND ');
+      }
+
+      const url = await discoverLocator.getRedirectUrl({
+        indexPatternId: dataViewId,
+        refreshInterval: {
+          display: 'Off',
+          pause: true,
+          value: 0,
+        },
+        timeRange: {
+          from,
+          to,
+          mode: 'absolute',
+        },
+        query: {
+          language: 'kuery',
+          query: kqlQuery,
+        },
+        // interval,
```

Comment:
do we need this comment?

Reply:
Removed here https://github.com/elastic/kibana/pull/120450/commits/877fb6b603705d8b31271728f1e4935a4d1a7835